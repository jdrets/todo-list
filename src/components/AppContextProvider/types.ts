import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
