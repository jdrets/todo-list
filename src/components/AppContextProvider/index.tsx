import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
