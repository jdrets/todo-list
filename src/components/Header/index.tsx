import React from 'react'

import {
  HeaderWrapper
} from './styles'

const Header = () => {

  return (
    <HeaderWrapper>
      sa
    </HeaderWrapper>
  )
}

export default Header