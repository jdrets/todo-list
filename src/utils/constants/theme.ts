const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
