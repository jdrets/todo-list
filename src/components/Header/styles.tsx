import styled from 'styled-components'

import theme from '../../utils/constants/theme';

export const HeaderWrapper = styled.header`
  background-color: ${props => props.theme.colors.primaryColor};
  color: white;
  height: 210px;
  padding: 0 12px;
`

export const DateLabel = styled.section`
  padding: 24px 0;
  text-align: center;
`

export const Container = styled.section`
  display: flex;
  justify-content: space-between;
`

export const DataWrapper = styled.div`
  text-align: left;
`

export const Title = styled.h1`
  font-size: 24px;
  font-weight: 900;
`

export const TasksCountLabel = styled.h2`
  font-size: 14px;
  font-weight: 500;
`

export const ButtonWrapper = styled.div`
  align-items: center;
  display: flex;
`
