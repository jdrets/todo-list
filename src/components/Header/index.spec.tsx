import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render, fireEvent } from '@testing-library/react'

import Header from '.'
import theme from '../../utils/constants/theme'
import AppContextProvider from '../AppContextProvider'

const setup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <Header/>
      </AppContextProvider>
    </ThemeProvider>
  )
}

describe('<Header />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })

  test('should call console.log when button is clicked', () => {
    const consoleLogMock = jest.fn()
    const { getByText } = setup()
    const button = getByText('Add new')

    console.log = consoleLogMock

    fireEvent.click(button)

    expect(consoleLogMock).toMatchSnapshot()
  })
})
