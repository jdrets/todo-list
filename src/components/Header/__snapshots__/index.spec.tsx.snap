// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match with snapshot 1`] = `
<div>
  <header
    class="sc-gtssRu dokrQo"
  >
    <section
      class="sc-dlnjPT dqIPfq"
    >
      6 July
    </section>
    <section
      class="sc-hKFyIo eYbNZd"
    >
      <div
        class="sc-eCApGN eaCnDs"
      >
        <h1
          class="sc-jSFkmK dqRofX"
        >
          To do list
        </h1>
        <h2
          class="sc-gKAblj btqBcj"
        >
          0
           tasks
        </h2>
      </div>
      <div
        class="sc-iCoHVE gCTGnh"
      >
        <div
          class="sc-bdnylx gcRwmt"
        >
          Add new
        </div>
      </div>
    </section>
  </header>
</div>
`;

exports[`<Header /> should match with snapshot when button is not allowed 1`] = `
<div>
  <header
    class="sc-gtssRu dokrQo"
  >
    <section
      class="sc-dlnjPT dqIPfq"
    >
      6 July
    </section>
    <section
      class="sc-hKFyIo eYbNZd"
    >
      <div
        class="sc-eCApGN eaCnDs"
      >
        <h1
          class="sc-jSFkmK dqRofX"
        >
          To do list
        </h1>
        <h2
          class="sc-gKAblj btqBcj"
        >
          0
           tasks
        </h2>
      </div>
    </section>
  </header>
</div>
`;
